piq/vsi.py

```diff
@@ -0,0 +1,285 @@
+            target: Batch of target images with shape  (batch_size x channels x H x W)
+            reduction: Reduction over samples in batch: "mean"|"sum"|"none"
+            data_range: Value range of input images (usually 1.0 or 255). Default: 1.0
+            c1: coefficient to calculate saliency component of VSI
+            c2: coefficient to calculate gradient component of VSI
+            c3: coefficient to calculate color component of VSI
+            alpha: power for gradient component of VSI
+            beta: power for color component of VSI
+            omega_0: coefficient to get log Gabor filter at SDSP
+            sigma_f: coefficient to get log Gabor filter at SDSP
+            sigma_d: coefficient to get SDSP
+            sigma_c: coefficient to get SDSP
+
+        Returns:
+            VSI: Index of similarity between two images. Usually in [0, 1] interval.
+        Note:
+            See https://ieeexplore.ieee.org/document/6873260 for details.
+        """
+
+    _validate_input(input_tensors=(prediction, target), allow_5d=False)
+    prediction, target = _adjust_dimensions(input_tensors=(prediction, target))
+    assert prediction.size(-3) == 3 and target.size(-3) == 3, 'Expected RGB images, got images with ' \
+                                                              '{prediction.size(-3)} and {target.size(-3)} channels'
+
+    prediction = prediction * 255. / data_range
+    target = target * 255. / data_range
+
+    vs_prediction = sdsp(prediction, data_range=255., omega_0=omega_0,
+                         sigma_f=sigma_f, sigma_d=sigma_d, sigma_c=sigma_c)
+    vs_target = sdsp(target, data_range=255., omega_0=omega_0, sigma_f=sigma_f,
+                     sigma_d=sigma_d, sigma_c=sigma_c)
+
+    # Weitghts to translate from RGB colour space to LMN, see https://ieeexplore.ieee.org/document/6873260
+    weights_RGB_to_LMN = torch.tensor([[0.06, 0.63, 0.27],
+                                       [0.30, 0.04, -0.35],
+                                       [0.34, -0.6, 0.17]]).to(target)
+
+    prediction_LMN = torch.matmul(prediction.permute(0, 2, 3, 1), weights_RGB_to_LMN.t()).permute(0, 3, 1, 2)
+    target_LMN = torch.matmul(target.permute(0, 2, 3, 1), weights_RGB_to_LMN.t()).permute(0, 3, 1, 2)
+
+    kernel_size = max(1, round(min(vs_prediction.size()[-2:]) / 256))
+    vs_prediction = avg_pool2d(vs_prediction.unsqueeze(1), kernel_size=kernel_size).squeeze(1)
+    vs_target = avg_pool2d(vs_target.unsqueeze(1), kernel_size=kernel_size).squeeze(1)
+
+    prediction_LMN = avg_pool2d(prediction_LMN, kernel_size=kernel_size)
+    target_LMN = avg_pool2d(target_LMN, kernel_size=kernel_size)
+
+    gm_prediction = scharr_grad_map(prediction_LMN[:, 1].unsqueeze(1)).squeeze(1)
+    gm_target = scharr_grad_map(target_LMN[:, 1].unsqueeze(1)).squeeze(1)
+
+    s_vs = (2 * vs_prediction * vs_target + c1) / (vs_prediction.pow(2) + vs_target.pow(2) + c1)
```

Comment:
This type of similarity computation is also very common. SSIM, MS-SSIM, GMSD, VIF and many other not yet implemented metrics use it
How about doing something like this:
```
def similarity_map(map1, map2, constant):
     return (2 * map1 * map2 + constant) / (map2.pow(2) + map2.pow(2) + constant)
```

```suggestion
    s_vs = similarity_map(vs_prediction, vs_target, c1)
```